Sources/Util/DynamicShadowView.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  Copyright © FINN.no AS, Inc. All rights reserved.
+//
+
+import UIKit
+
+public class DynamicShadowView: UIView {
+    static let maxShadowRadius: CGFloat = 3
+
+    // MARK: - Init
+
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        setup()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        setup()
+    }
+
+    // MARK: - Shadow
+
+    public func updateShadow(using scrollView: UIScrollView) {
+        let contentFrame = CGRect(
+            x: -scrollView.contentOffset.x,
+            y: scrollView.frame.minY - scrollView.contentOffset.y - scrollView.contentInset.top,
+            width: scrollView.contentSize.width,
+            height: scrollView.contentSize.height + scrollView.contentInset.top
+        )
+
+        let intersection = contentFrame.intersection(frame)
+        layer.shadowRadius = min(intersection.height * 0.2, DynamicShadowView.maxShadowRadius)
+    }
+
+    // MARK: - Setup
+
+    private func setup() {
+        backgroundColor = .white
```

Comment:
is this intentional?

and btw, why is this class used?

Reply:
Nope, will change it to `bgPrimary` 👍 This class is used when we need to update shadow appearance based on the content offset of the scroll view. It's a header view in most cases, but this time I also needed it as a footer view, so I had to modify existing `BottomShadowView` to have the shadow not only on bottom, but on top as well.